packages/editor-ui/src/stores/nodeCreator.ts

```diff
@@ -305,31 +305,40 @@ export const useNodeCreatorStore = defineStore(STORES.NODE_CREATOR, {
 			return nodesWithActions;
 		},
 		mergedAppNodes(): INodeTypeDescription[] {
-			const mergedNodes = this.visibleNodesWithActions.reduce(
-				(acc: Record<string, INodeTypeDescription>, node: INodeTypeDescription) => {
-					const clonedNode = deepCopy(node);
-					const isCoreNode = node.codex?.categories?.includes(CORE_NODES_CATEGORY);
-					const actions = node.actions || [];
-					// Do not merge core nodes
-					const normalizedName = isCoreNode
-						? node.name
-						: node.name.toLowerCase().replace('trigger', '');
-					const existingNode = acc[normalizedName];
-
-					if (existingNode) existingNode.actions?.push(...actions);
-					else acc[normalizedName] = clonedNode;
-
-					if (!isCoreNode) {
-						acc[normalizedName].displayName = node.displayName.replace('Trigger', '');
-					}
-
-					acc[normalizedName].actions = filterSinglePlaceholderAction(
-						acc[normalizedName].actions || [],
-					);
-					return acc;
-				},
-				{},
-			);
+			const mergedNodes = [...this.visibleNodesWithActions]
+				// Sort triggers so they are always on top and when later get merged
+				// they won't be disacrded if they have the same name as a core node which doesn't contain actions
+				.sort((a, b) => {
```

Comment:
should we move this sort to `filteredMergedAppNodes` in `TriggerHelperPanel.vue` instead? that way the sort happens only before the actual rendering.

Reply:
The sort needs to happen before the nodes are merged. But in `filteredMergedAppNodes` we're already using merged nodes to do the filtering.